columnq/src/table/mod.rs

```diff
@@ -473,15 +473,15 @@ impl TableSource {
     }
 }
 
-pub async fn load(t: &TableSource) -> Result<Arc<dyn TableProvider>, ColumnQError> {
+pub async fn load(t: &TableSource, dfctx: &datafusion::execution::context::SessionContext) -> Result<Arc<dyn TableProvider>, ColumnQError> {
```

Comment:
@houqp passing `dfctx` seems not ideal, because it is only used for a few TableLoadOptions, e.g. parquet (this PR), maybe csv in future.

What do you think? Let me know if you have other ideas

Reply:
yeah, I think this is unavoidable unfortunately, we will likely use it for most of the formats supported by datafusion including csv and json, so it's not too bad.

Reply:
sounds good